assume/common/scenario_loader.py

```diff
@@ -254,7 +254,8 @@ async def load_scenario_folder_async(
 
     # load the config file
     path = f"{inputs_path}/{scenario}"
-    config = yaml.safe_load(open(f"{path}/config.yaml", "r"))
+    with open(f"{path}/config.yaml", "r") as f:
```

Comment:
just for my knowledge, what is the difference between these two approaches?

Reply:
when using open() in a single the file handler i opened but never actually closed, so you are relying on the garbage collector for closing it.

By using `with open(...) as f:` you are making sure that the file handler is actually closed directly afterwards.
It is best to interact with file ressources as short as possible and clean it up as soon as possible.

https://stackoverflow.com/a/7409814